### ISSUE TITLE
[Dialog|A11Y]: Extra unexpected data is read by screenreaders inside Dialog window

### ISSUE THREAD

**Issue by MaksymZahreba** [UI5/webcomponents#13562]
### Bug Description

When a modal dialog is opened JAWS screen reader announces additional, irrelevant information such as "Header toolbar. To move through the toolbar use left or right arrow," which is not part of the modal dialog content. This can confuse users and detract from the intended user experience.

<img width="888" height="656" alt="Image" src="https://github.com/user-attachments/assets/ebc42ae9-72b2-4599-95f5-60ca601e5c95" />

### Affected Component

Dialog

### Expected Behaviour

Expected result:

Upon opening the modal dialog, JAWS should announce only the modal dialog content (e.g., modal title and the first interactive element). No information from outside the modal (such as toolbars or headers) should be announced.

WCAG/BP: 4.1.2 Name, Role, Value

Help links/Techniques: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/

Recommendation: According to MDN (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/toolbar_role), a toolbar is a collection of commonly used program controls. A modal header containing a title and a close button does not fit this definition. role="toolbar" with aria-label="Header" misleads screen readers. It should be removed.

### Steps to Reproduce

1. open Dialog component
2. launch screenreader
3. navigate through Dialog window
...

### Priority

Medium

### UI5 Web Components Version

2.22.0

### Browser

Chrome, Edge, Firefox

### Operating System

Windows 10\11

### Organization

SAP \ UDEx

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by IlianaB** [comment]
Hello @MaksymZahreba ,
in order for me to understand the issue correctly, could you please edit this [sample](https://ui5.github.io/webcomponents/play/#eNq9Vm1v2zYQ_iusP9lBJCXYUgyyHTQvDlCgWIO1_TDM-0CLZ4sNRaok5cQL_N93R1G2nDVL1g9Dg1q8l-ceHu-OfBxILeAhLX2lBvnjQPMKBnlfeDwojPagPYonb5KE1YpvVtY0WiRLowRLkvO5nry5_nj1-ffbGSMnEtAvU1yvpvMB6PkAZe2_SQlc4Irh38T5jQJaHOXa-GEuYCk1iBF7RJmQjoLlTBsNYxRsyT9ECPqlsQWIpDDK2ISLr43zB7ZsknX4_3-4SQWes6Lk1oHHHHz5fJP8EtIQVZRqlK8l3NfG-vmAxUSj8F4KX04FrGUBSVgcM6mll1wlruAKpqfpSQTz0mPMT7yqFUyydkVyJfUds6AQLtByJQBFKS0sUZZmFZc6LZwLOJMsHgt-LozYsOCDdgte3MXjDjvP2ZrbYZI4Xl_uVFekGY1ZCXJVYlZ-PjupH8aR4HeKJgEdCwfDNfIsWTTeG82kwIgCd2lWH2vQYBGCPth1kE2yve3et7Xv-9IucTdgEw8PlM_fYCWdB8tujK0iLQeFlxizUNw5tEE_qZHdzkDIdfilGIovQFEFoGHjwNLhYRQL3xppQTBXmvuQH33-JapbssFxByN13fiW6R7lvLUMumCZdZGfY1Aj43tjxTMMbqP6BQY9FL-p6axv95I1Vw0kznNPil9hxb1c_xBZwDpT-xizuPwe8aB7gXWE--88uBAWqNr7ES9a4Qsxd67PR81iOe3cF9wypwxV39IYT6XMBDi50ii56SR781DUuIo-QE0VxsF8MN8XaVvfV8o4cqeh1UHOqrrkTv4FIsgJ-BoKHAIQ6B00TrfmNnT-voni6Hp2zGPXFFbWvjvMyohGUSM4W9CSJsrXNk-tYcCnedK2a9bdDv86Ewbb4w6qdyd1kv6FlB31UUopgB1lcy0rmqiMhlxP7ZqaxIEgjux_-gYGB_7vMDXZPSwKgxKNMR0euPNZO446pBfNL0PiX23-nsrr1dYfcM6_3phK_PW8uQ22dHY49Vl_MrMpE6ZoKrROV-BnCujzcvNeDJ-O8NG47_4Kx6cubb079PwjTdNWln5rwG4-gcK-M_ZCKfROD9tj9OeYmPfJpLwosJPlQirpNxfeW4ldAQRNVzx20K2pmzpneyrH1H5YcdYol0dCqRQo3o6fggsxW-NmPtBlgwKkhB1Y3CEIG47Y9Dy-MwKEoXttyrxt6AGxHe3A4m5THFszXpTDhdedK37-UJAlV659poxCrF2H4fX_tMVI1O-xdH8xHj6TlgoeCJN-cX7YdgDmmCzVVJo09ESSy00SwXLmao4PGkD6akMGXOHwSqSHyrV4dONYT6qKWwybs58sVOwt_kfE6Tj7fM6RzfqQ1cpKQf7h1ZSz07PWNeIlC4OtWOUsPetBvqsA8zWs-EMS3d6e4AOmfRY-3f8BsdOIjjDbwXb7N9gC9lo) to demonstrate the structure of your Dialog?
Could you please confirm you use ui5-bar in the footer with one button (as in the sample above)?
And what about the header?

Thanks!

**Comment by MaksymZahreba** [comment]
Hi @IlianaB,
in order to avoid cases of component use being the case for issue reproduction I have not created any new Dialog nor applied any edits to it and used existing example in UI5 library
https://ui5.github.io/webcomponents/components/Dialog/#usage-of-bar-as-headerfooter

**Comment by IlianaB** [comment]
Thank you @MaksymZahreba , that is what I needed - the exact sample/set up you use.

**Comment by IlianaB** [comment]
Hello @UI5/ui5-team-ctr-bal ,
I am forwarding this issue to you, because the problem described in this issue comes from the ui5-bar component. According to Accessibility Specification, role "toolbar" should not be present when Bar has less than two interactive elements (same as in OpenUI5).
Use sample: https://ui5.github.io/webcomponents/components/Dialog/#usage-of-bar-as-headerfooter

Regards,
Iliana

**Comment by PlutaKatarzyna** [comment]
Hi @IlianaB is there any ETA for this ticket?

**Comment by unazko** [comment]
Hello @MaksymZahreba,

The documentation was confusing recommending to the users to set the `accessibleRole` property to `Toolbar` whenever there are more than two interactive elements in the contents. In reality the users should do the opposite, which is to set the `accessibleRole` property to `None` when there are less than two interactive elements in the slot as the default value is `Toolbar` for this property. If there are more than two interactive elements in the contents then it's not needed to set `accessibleRole` prop to `Toolbar` as it is already pre-set.

We've improved the documentation and the playground samples via the following PR:
https://github.com/UI5/webcomponents/pull/13706

We can't directly fix this issue though as our fix will be backward incompatible.
For instance the default value for the `accessibleRole` property is `Toolbar`.

https://ui5.github.io/webcomponents/components/Dialog/#usage-of-bar-as-headerfooter

In cases like the provided one with less than one interactive element in the `ui5-bar` contents you would need to set explicitly
the `accessibleRole` property to `None` so the "toolbar" isn't set:

```html
<ui5-bar slot="footer" accessible-role="None">
    <ui5-button class="dialogCloser" design="Emphasized" slot="endContent"
        style="min-width: 4rem;">OK</ui5-button>
</ui5-bar>
```

And for cases with more than two interactive elements in the contents slot the application developers do not need to set
the `accessibleRole` property as it will have a default value of "Toolbar":

```html
<ui5-bar design="Header">
      <ui5-button icon="home" tooltip="Go home"  slot="startContent"></ui5-button>
      <ui5-label id="basic-label">Content</ui5-label>
      <ui5-button icon="action-settings" tooltip="Go to settings" slot="endContent"></ui5-button>
</ui5-bar>
```

Best regards,
Boyan
